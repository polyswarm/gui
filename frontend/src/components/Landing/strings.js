const strings = {
  title: 'PolySwarm Relay',
  go: 'Go!'
};

export default strings;