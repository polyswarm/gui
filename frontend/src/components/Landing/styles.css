.Landing {
    display: flex;
    flex-direction: column;
    height: 100%;
    justify-content: center;
    left: 0;
    top: 0;
    widows: 100%;
}

.Landing-Content {
    color: white;
    font-size: 2em;
    margin: 0 auto;
    width: 80%;
    z-index: 2;
}

.Landing-Background {
    background-color: #8500ff;
    display: block;
    height: 100vh;
    left: 0;
    position: absolute;
    top: 0;
    width: 100vw;
    z-index: 1;
}

.Landing-nectar {
    display: flex;
    flex-direction: row;
}

.Landing-nectar > .NectarField {
    flex-grow: 8
}

.Landing-nectar > .Button {
    margin-top: auto;
    height: 48px;
}

.translate-enter {
    transform: translateY(-46vh);
}

.translate-enter-active {
    transform: translateY(0);
    transition: all 300ms ease-out;
}

.translate-exit {
    transform: translateY(0);
}

.translate-exit-active {
    transform: translateY(-46vh);
    transition: all 300ms ease-out;
}

.translate-exit-done {
    transform: translateY(-46vh);
}