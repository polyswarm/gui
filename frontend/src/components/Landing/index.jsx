// Vendor imports
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import {CSSTransition} from 'react-transition-group';
// Project imports
import NectarField from '../NectarField';
import Button from '../Button';
// Component imports
import strings from  './strings.js';
import './styles.css';

class Landing extends Component {
  constructor(props) {
    super(props);
    this.state = {
        nectar: '',
        error: false,
        hide: false
    };

    this.onNectarChanged = this.onNectarChanged.bind(this);
    this.onButtonClick = this.onButtonClick.bind(this);
  }

  render() {
    const {state: {error, nectar, hide, finished} } = this;
    return(
      <div className='Landing'>
        <div className='Landing-Background'/>
        <div className='Landing-Content'>
          <h1>{strings.title}</h1>
          <div className='Landing-nectar'>
            <CSSTransition
              in={!hide}
              classNames='translate'
              onExited={() => {
                const {props: {onSetNectar}, state: {nectar}} = this;
                if (onSetNectar) {
                  onSetNectar(nectar);
                }
              }}
              timeout={300}>
              {() => (
                <NectarField onChange={this.onNectarChanged}
                  nectar={nectar} />
              )}
            </CSSTransition>
            <Button
              disabled={error || nectar.length == 0}
              onClick={this.onButtonClick} >
              {strings.go}
            </Button>
          </div>
        </div>
      </div>
    );
  }

  onNectarChanged(nectar, valid) {
    this.setState({nectar: nectar, error: !valid});
  }

  onButtonClick() {
    this.setState({hide: true});
  }

}
Landing.proptypes = {
  onSetNectar: PropTypes.func
}
export default Landing;