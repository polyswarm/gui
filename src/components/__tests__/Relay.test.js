import React from 'react';
import {render, shallow, mount} from 'enzyme';
import {renderToJson} from 'enzyme-to-json';
import Landing from '../Landing';

const chain = {name:'asdf', balance: 1};

it('renders without crashing', () => {
  const wrapper = render(<Landing homechain={chain} sidechain={chain}/>);
  expect(renderToJson(wrapper)).toMatchSnapshot();
});

it('calls onNectarChanged when NectarField is changed', () => {
  const onNectarChanged = jest.spyOn(Landing.prototype, 'onNectarChanged');
  const wrapper = mount(<Landing homechain={chain} sidechain={chain}/>);
  wrapper.find('.NectarField').find('input').simulate('change', {target: {value: '1'}});

  expect(onNectarChanged).toHaveBeenCalledWith('1', true);
});

it('disables the button when nectar is negative', () => {
  const wrapper = mount(<Landing homechain={chain} sidechain={chain}/>);
  wrapper.find('.NectarField').find('input').simulate('change', {target: {value: '-1'}});

  expect(wrapper.find('.Button').props().disabled).toBeTruthy();
});

it('enables the Go button when nectar meets length requirements in NectarField', () => {
  const wrapper = mount(<Landing homechain={chain} sidechain={chain}/>);
  wrapper.find('.NectarField').find('input').simulate('change', {target: {value: '1'}});

  expect(wrapper.find('.Button').props().disabled).toBeFalsy();
});

it('disables the Go button when nectar length is 0', () => {
  const wrapper = mount(<Landing homechain={chain} sidechain={chain}/>);
  wrapper.find('.NectarField').find('input').simulate('change', {target: {value: ''}});

  expect(wrapper.find('.Button').props().disabled).toBeTruthy();
});

it('calls setstate with the nectar when nectar modified', () => {
  const setState = jest.spyOn(Landing.prototype, 'setState');
  const wrapper = mount(<Landing homechain={chain} sidechain={chain}/>);

  wrapper.find('.NectarField').find('input').simulate('change', {target: {value: '1'}});

  expect(setState).toHaveBeenCalledWith({nectar: '1', error: false});
});

it('calls setstate with the nectar & error true when nectar modified with invalid value', () => {
  const setState = jest.spyOn(Landing.prototype, 'setState');
  const wrapper = mount(<Landing homechain={chain} sidechain={chain}/>);
  setState.mockClear();
  wrapper.find('.NectarField').find('input').simulate('change', {target: {value: '0'}});

  expect(setState).toHaveBeenCalledWith({nectar: '0', error: true});
});

it('calls onTransfer button is clicked', () => {
  const onTransfer = jest.fn();
  const wrapper = mount(<Landing homechain={chain} sidechain={chain} onTransfer={onTransfer} />);
  wrapper.setState({nectar: '0123456789123456'});
  wrapper.find('button').simulate('click');

  expect(onTransfer).toHaveBeenCalledWith('0123456789123456', 0);
});

it('updates the after field balances when nectar amount entered', () => {
  const wrapper = mount(<Landing homechain={chain} sidechain={chain} />);
  wrapper.setState({nectar: .5});

  expect(wrapper.find('.ChainInfo').last().find('li').first().text()).toEqual('asdf: 0.5');
  expect(wrapper.find('.ChainInfo').last().find('li').last().text()).toEqual('asdf: 1.5');
});

it('Changes the balances when selected index is changed', () => {
  const wrapper = mount(<Landing homechain={chain} sidechain={chain} />);
  wrapper.setState({nectar: .5, selected: 1});

  expect(wrapper.find('.ChainInfo').last().find('li').first().text()).toEqual('asdf: 1.5');
  expect(wrapper.find('.ChainInfo').last().find('li').last().text()).toEqual('asdf: 0.5');
});

it('Disbales the button when error set to true', () => {
  const wrapper = mount(<Landing homechain={chain} sidechain={chain} />);
  wrapper.setState({error: true});

  expect(wrapper.find('.Button').props().disabled).toBeTruthy();
});

it('Disables the button when nectar is greater than the homechain balance in deposit', () => {
  const setState = jest.spyOn(Landing.prototype, 'setState');
  const wrapper = mount(<Landing homechain={chain} sidechain={chain} />);
  wrapper.setState({selected: 0});
  const instance = wrapper.instance();
  setState.mockClear();

  instance.onNectarChanged(2, true);

  expect(setState).toHaveBeenCalledWith({nectar: 2, error: true});
});

it('Disables the button when nectar is greater than the sidechain balance in withdrawal', () => {
  const setState = jest.spyOn(Landing.prototype, 'setState');
  const wrapper = mount(<Landing homechain={chain} sidechain={chain} />);
  wrapper.setState({selected: 1});
  const instance = wrapper.instance();

  instance.onNectarChanged(2, true);

  expect(setState).toHaveBeenCalledWith({nectar: 2, error: true});
});