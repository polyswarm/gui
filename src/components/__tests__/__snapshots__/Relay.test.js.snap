// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  class="Relay"
>
  <header
    class="Header"
  >
    <div
      class="Header-Row"
    >
      <div
        class="Header-Title"
      >
        <img
          alt="PolySwarm logo"
          class="Header-Logo"
          src="../public/img/back-arrow.svg"
        >
          
        </img>
        <h3>
          PolySwarm Relay
        </h3>
        <span
          class="RequestSpinner"
        >
          
        </span>
      </div>
      <div
        class="Header-Actions"
      >
        
      </div>
    </div>
    <div
      class="Header-Address"
    >
      <p
        class="Header-Divided"
        style="font-size:7.0588235294117645vw"
      >
        author
      </p>
      <p
        class="Header-Divided"
        style="font-size:7.0588235294117645vw"
      >
        1 NCT 
      </p>
      <p
        style="font-size:7.0588235294117645vw"
      >
        1 ETH
      </p>
    </div>
  </header>
  <div
    class="Relay-Content"
  >
    <div
      class="Relay-Centered"
    >
      <div
        class="Relay-Chain"
      >
        <div
          class="ChainInfo"
        >
          <div
            class="ChainInfo-Title"
          >
            <h3>
              Current
            </h3>
          </div>
          <div
            class="ChainInfo-Content"
          >
            <p>
              
            </p>
            <ul>
              <li
                class="StatRow-Vertical"
              >
                <p
                  class="StatTitle"
                >
                  Main Ethereum Chain
                </p>
                <p
                  class="StatContent"
                >
                  1 NCT
                </p>
              </li>
              <li
                class="StatRow-Vertical"
              >
                <p
                  class="StatTitle"
                >
                  PolySwarm Side Chain
                </p>
                <p
                  class="StatContent"
                >
                  1 NCT
                </p>
              </li>
            </ul>
          </div>
        </div>
        <div
          class="ChainInfo"
        >
          <div
            class="ChainInfo-Title"
          >
            <h3>
              Preview
            </h3>
          </div>
          <div
            class="ChainInfo-Content"
          >
            <p>
              
            </p>
            <ul>
              <li
                class="StatRow-Vertical"
              >
                <p
                  class="StatTitle"
                >
                  Main Ethereum Chain
                </p>
                <p
                  class="StatContent"
                >
                  1 NCT
                </p>
              </li>
              <li
                class="StatRow-Vertical"
              >
                <p
                  class="StatTitle"
                >
                  PolySwarm Side Chain
                </p>
                <p
                  class="StatContent"
                >
                  1 NCT
                </p>
              </li>
            </ul>
          </div>
        </div>
      </div>
      <div
        class="Relay-Nectar"
      >
        <div
          class="RelayDropdown"
        >
          <p
            class="RelayDropdown-Selection"
          >
            Deposit
          </p>
          <div
            class="RelayDropdown-Choices"
          >
            <p>
              Deposit
            </p>
            <p>
              Withdraw
            </p>
          </div>
        </div>
        <div
          class="AnimatedInput"
        >
          <div>
            
          </div>
          <input
            class="AnimatedInput-Input AnimatedInput-Error"
            id="nectar"
            type="number"
            value=""
          >
            
          </input>
          <p
            class="AnimatedInput-ErrorLabel"
          >
            This feature is disabled until the PolySwarm Sidechain goes live very soon.
          </p>
        </div>
      </div>
      <div
        class="Relay-Button"
      >
        <button
          class="Button"
          disabled=""
        >
          Transfer!
        </button>
      </div>
    </div>
  </div>
</div>
`;
